backend/cn/main.ml

```diff
@@ -499,12 +501,58 @@ let () =
       Verify_flags.use_peval $
       Verify_flags.batch $
       Common_flags.no_inherit_loc $
-      Common_flags.magic_comment_char_dollar $
-      Test_Generation_Flags.output_test_dir $
-      Test_Generation_Flags.test_max_unfolds $
-      Test_Generation_Flags.testing_framework
+      Common_flags.magic_comment_char_dollar
+  in
+  let info = Cmd.info "verify" in
+  Cmd.v info verify_t
+
+module Test_generation_flags = struct
+let output_test_dir =
+  let doc = "[Experimental] Place generated test suites in the provided directory" in
+  Arg.(required & opt (some string) None & info ["output-dir"] ~docv:"FILE" ~doc)
+
+let test_max_unfolds =
+  let doc = "[Experimental] Set the maximum number of unfolds for recursive predicates" in
+  Arg.(value & opt int 5 & info ["max-unfolds"] ~doc)
+
+let testing_framework =
+  let doc = "[Experimental] Specify the testing framework used by generated tests" in
+  Arg.(value & opt (enum ["gtest", TestGeneration.GTest]) GTest & info ["framework"] ~doc)
+
+end
+
+let generate_tests_cmd =
+  let open Term in
+  let generate_tests_t =
+    const generate_tests $
+    Common_flags.file $
+    Common_flags.macros $
+    Common_flags.incl_dirs $
+    Common_flags.incl_files $
+    Common_flags.astprints $
+    Verify_flags.use_peval $
+    Common_flags.no_inherit_loc $
+    Common_flags.magic_comment_char_dollar $
+    Test_generation_flags.output_test_dir $
+    Test_generation_flags.test_max_unfolds $
+    Test_generation_flags.testing_framework
   in
+  let doc = String.trim
+    "
+    Generates RapidCheck tests for all functions in [FILE] with CN specifications.
+    The tests use randomized inputs, which are guaranteed to satisfy the CN precondition.
+    A [.cpp] file containing the test harnesses will be placed in [output-dir].
+    "
+  in
+  let info = Cmd.info "generate-tests" ~doc:doc in
+  Cmd.v info generate_tests_t
+
+let subcommands = [
+  verify_cmd;
+  generate_tests_cmd;
+]
+
+let () =
   let version_str = "CN version: " ^ Cn_version.git_version in
   let cn_info = Cmd.info "cn" ~version:version_str in
-  Stdlib.exit @@ Cmd.(eval (v cn_info check_t))
-
+  Stdlib.exit @@ Cmd.(eval (group cn_info subcommands))
```

Comment:
I think if you put `verify_cmd` as the default instead of `cn_info` this should retain backwards compatibility - with a deprecation warning and suggestion to use `verify`.
You can keep `verify_cmd` in the subcommands list so that both work, and we can transition the CI smoothly.

Reply:
That makes sense